src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,301 @@
+        $ids = $this->finSearchConfigRepository->searchIds($criteria, Context::createDefaultContext())->getIds();
+
+        return array_shift($ids);
+    }
+
+    private function buildCriteria(string $salesChannelId, string $languageId, ?string $key = null): Criteria
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(new EqualsFilter('salesChannelId', $salesChannelId));
+        $criteria->addFilter(new EqualsFilter('languageId', $languageId));
+
+        if ($key) {
+            $criteria->addFilter(new EqualsFilter('configurationKey', $key));
+        }
+
+        return $criteria;
+    }
+
+    private function load(string $salesChannelId, string $languageId): array
+    {
+        $key = sprintf('%s-%s', $salesChannelId, $languageId);
+
+        if (isset($this->configs[$key])) {
+            return $this->configs[$key];
+        }
+
+        $criteria = new Criteria();
+        if (method_exists($criteria, 'setTitle')) {
+            $criteria->setTitle('finsearch-config::load');
+        }
+
+        $criteria->addFilter(
+            new MultiFilter(
+                MultiFilter::CONNECTION_AND,
+                [
+                    new EqualsFilter('salesChannelId', $salesChannelId),
+                    new EqualsFilter('languageId', $languageId),
+                ]
+            )
+        );
+
+        $criteria->addSorting(
+            new FieldSorting('salesChannelId', FieldSorting::ASCENDING),
+            new FieldSorting('id', FieldSorting::ASCENDING)
+        );
+        $criteria->setLimit(500);
+
+        /** @var FinSearchConfigCollection $configs */
+        $configs = $this->finSearchConfigRepository->search($criteria, Context::createDefaultContext())->getEntities();
+        $this->configs[$key] = $this->parseFindologicConfiguration($configs);
+
+        return $this->configs[$key];
+    }
+
+    private function buildFindologicConfig(FinSearchConfigCollection $configs): array
+    {
+        $findologicConfig = [];
+        foreach ($configs as $config) {
+            $keyExists = array_key_exists($config->getConfigurationKey(), $findologicConfig);
+            if (!$keyExists || !Utils::isEmpty($config->getConfigurationValue())) {
+                $findologicConfig[$config->getConfigurationKey()] = $config->getConfigurationValue();
+            }
+        }
+
+        return $findologicConfig;
+    }
+
+    /**
+     * The keys of the system configs look like `core.loginRegistration.showPhoneNumberField`.
+     * This method splits those strings and builds an array structure
+     * ```
+     * Array
+     * (
+     *     [core] => Array
+     *         (
+     *             [loginRegistration] => Array
+     *                 (
+     *                     [showPhoneNumberField] => 'someValue'
+     *                 )
+     *         )
+     * )
+     * ```
+     */
+    private function parseFindologicConfiguration(FinSearchConfigCollection $collection): array
```

Comment:
Isn't this parsing any configuration? If not the example is more confusing than it helps, if it is then rename the method.

Reply:
Technically yes, but in our service it will only parse Findologic config.

Reply:
Then please rename it to just `parseConfiguration`, since we are already in the `Findologic` context, since this service is already called `FindologicConfigService`.